File: app/src/main/java/org/test/kotlin/demo/repos/ReposContract.java
package org.test.kotlin.demo.repos;

import org.test.kotlin.demo.api.dto.RepositoryDTO;

import java.util.List;

import androidx.annotation.NonNull;

interface ReposContract {

    String ARG_USERNAME = "userName";

    interface View {

        void showRepositories(@NonNull List<RepositoryDTO> repositories);

    }

    interface Presenter {

        void onViewReady();

    }

}

File: app/src/main/java/org/test/kotlin/demo/repos/ReposContract.kt
package org.test.kotlin.demo.repos

import org.test.kotlin.demo.api.dto.RepositoryDTO

interface ReposContract {

    interface View {

        fun showRepositories(repositories: List<RepositoryDTO>)

        fun showError(throwable: Throwable)

    }

    interface Presenter {

        fun onViewReady()

    }

    companion object {

        const val ARG_USERNAME = "userName"

    }

}
